addons/mail/static/src/drag_and_drop/dropzone/dropzone.js

```diff
@@ -0,0 +1,59 @@
+/** @odoo-module **/
+
+import { Component, onMounted, useExternalListener, useRef, useState } from "@odoo/owl";
+
+
+export class Dropzone extends Component {
+    static template = "mail.dropzone";
+    static props = { ref: Object };
+
+    setup() {
+        // Prevents to browser to open or download the file when it is dropped
+        // outside of the dropzone.
+        useExternalListener(window, "dragover", ev => ev.preventDefault());
+        useExternalListener(window, "drop", ev => ev.preventDefault());
+
+        this.root = useRef("root");
+        this.state = useState({ isDraggingInside: false });
+
+        this._dragCount = 0;
+
+        onMounted(() => {
+            const { top, left, width, height } = this.props.ref.el.getBoundingClientRect();
+            Object.assign(this.root.el.style, {
+                top: `${top}px`,
+                left: `${left}px`,
+                width: `${width}px`,
+                height: `${height}px`,
+            });
+        });
+    }
+
+    /**
+     * @param {DragEvent} ev
+     */
+    onDragEnter(ev) {
+        if (this._dragCount === 0) {
+            this.state.isDraggingInside = true;
+        }
+        this._dragCount++;
+    }
+
+    /**
+     * @param {DragEvent} ev
+     */
+    onDragLeave(ev) {
+        this._dragCount--;
+        if (this._dragCount === 0) {
+            this.state.isDraggingInside = false;
+        }
+    }
+
+    /**
+     * @param {DragEvent} ev
+     */
+    onDrop(ev) {
+        this.state.isDraggingInside = false;
+    }
```

Comment:
I would not write docstring. It's not that helpful, and makes code harder to read at a glance. Notably we need to scroll more, which is annoying.